.travis/release.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -e
+
+usage() {
+    cat <<EOF
+    Usage: $0 DOCKERHUB_REPO JAR_VERSION"
+    Package (e.g. mvn package) project at version JAR_VERSION and push a Docker image to DOCKERHUB_REPO:JAR_VERSION.
+EOF
+}
+
+if [ $# -ne 2 ]; then
+  usage
+  exit 1
+fi
+
+dockerhub_repo="$1"
+jar_version="$2"
+image="$dockerhub_repo":"$jar_version"
+
+echo "Updating pom.xml <version> to $jar_version"
+mvn org.codehaus.mojo:versions-maven-plugin:2.7:set -DnewVersion=$jar_version
+
+echo "Packaging project"
+mvn package -DskipTests
+
+echo "Building Docker image $image"
+docker build -t $image --build-arg jar_version=$jar_version .
```

Comment:
It feels wrong to have the Maven build and Docker build happen as part of the Travis "deploy" stage.  Apart from anything else, this means that build errors will be reported as a build success with a deployment error, and that the build will not even be attempted for commits that don't match one of the deployment conditions.  Could the build be moved to the "script" stage instead, with only the "docker push" happening at the "deploy" stage?

Reply:
Good point, the "script" stage will run tests so test failures will be reported there but if `package` failed then you're right it would be reported as a deploy failure.

My only concern is ensuring the correct version is used both when packaging and tagging the image - this will require a conditional in the "script" to set the version on a tagged push but I don't think it should be an issue 👍